### PR TITLE
enhancement(antithesis): DogStatsD load generation

### PR DESCRIPTION
**Pull request by blt**
## Summary

This PR introduces DogStatD load generation. If you're familiar with
my work in lading you might notice this is very different. The
implementation here is responsible for giving Antithesis choices between
'compliant' dogstatsd lines and 'feral' lines which are _mechanically_
allowable lines but are in practice not sensible. Also of interest
control of contexts and send rates are jettisoned entirely, that's an
antithesis scheduling concern.

Metrics, events and service-checks are all present. Depending on the
antithesis shots I may need to adjust the ratios of compliant to feral
outcomes.

## Change Type
- [ ] Bug fix
- [ ] New feature
- [x] Non-functional (chore, refactoring, docs)
- [ ] Performance

## How did you test this PR?

Antithesis shots confirmed load generation behaves like a feral animal.

### CODE REVIEW

**Comment by blt** [comment]
* **#1799** <a href="https://app.graphite.com/github/pr/DataDog/saluki/1799?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2  dependent PRs ([#1800](https://github.com/DataDog/saluki/pull/1800) <a href="https://app.graphite.com/github/pr/DataDog/saluki/1800?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#1826](https://github.com/DataDog/saluki/pull/1826) <a href="https://app.graphite.com/github/pr/DataDog/saluki/1826?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#1905** <a href="https://app.graphite.com/github/pr/DataDog/saluki/1905?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1785** <a href="https://app.graphite.com/github/pr/DataDog/saluki/1785?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1782** <a href="https://app.graphite.com/github/pr/DataDog/saluki/1782?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/DataDog/saluki/1782?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#1779** <a href="https://app.graphite.com/github/pr/DataDog/saluki/1779?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1768** <a href="https://app.graphite.com/github/pr/DataDog/saluki/1768?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2 other dependent PRs ([#1769](https://github.com/DataDog/saluki/pull/1769) <a href="https://app.graphite.com/github/pr/DataDog/saluki/1769?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#1787](https://github.com/DataDog/saluki/pull/1787) <a href="https://app.graphite.com/github/pr/DataDog/saluki/1787?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#1767** <a href="https://app.graphite.com/github/pr/DataDog/saluki/1767?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1774** <a href="https://app.graphite.com/github/pr/DataDog/saluki/1774?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: test/antithesis/harness/src/payload/dogstatsd/common.rs [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+        }
+    }
+}
+
+/// Sample a count of segments and join them with sampled `separators`. A pool of
+/// `N` segments over a count `c` gives `N^c` results.
+pub(crate) fn write_segments<R: Rng + ?Sized>(
+    rng: &mut R, buf: &mut Vec<u8>, vibe: Vibe, compliant: &[&[u8]], aberrant: &[&[u8]], separators: &[u8],
+) {
+    let count = Boundary::<u8>::new().sample(rng);
```
[the remark this answers is not in the document]
Comment (blt) [reply test/antithesis/harness/src/payload/dogstatsd/common.rs:137]: An accepted edge-case.

[the remark this answers is not in the document]
Comment (blt) [reply test/antithesis/harness/src/payload/dogstatsd/common.rs:137]: An accepted edge-case.

[the remark this answers is not in the document]
Comment (blt) [reply test/antithesis/harness/src/payload/dogstatsd/common.rs:137]: An accepted edge-case.

[the remark this answers is not in the document]
Comment (blt) [reply test/antithesis/harness/src/payload/dogstatsd/common.rs:137]: An accepted edge-case.

File: test/antithesis/harness/src/payload/dogstatsd/common.rs [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+            }
+        }
+        extend_choice(buf, vibe, compliant, aberrant);
+    }
+}
+
+/// An identifier (name, host, key, source) built from word segments.
+pub(crate) fn write_words<R: Rng + ?Sized>(rng: &mut R, buf: &mut Vec<u8>, vibe: Vibe) {
+    write_segments(rng, buf, vibe, COMPLIANT_WORD, ABERRANT_WORD, NAME_SEPARATORS);
+}
```
[the remark this answers is not in the document]
Comment (blt) [reply test/antithesis/harness/src/payload/dogstatsd/common.rs:151]: An accepted edge-case.

File: test/antithesis/harness/src/payload/dogstatsd/common.rs [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+    }
+}
+
+/// Sample a count of segments and join them with sampled `separators`. A pool of
+/// `N` segments over a count `c` gives `N^c` results.
+pub(crate) fn write_segments<R: Rng + ?Sized>(
+    rng: &mut R, buf: &mut Vec<u8>, vibe: Vibe, compliant: &[&[u8]], aberrant: &[&[u8]], separators: &[u8],
+) {
+    let count = Boundary::<u8>::new().sample(rng);
+    for i in 0..count {
```
[the remark this answers is not in the document]
Comment (blt) [reply test/antithesis/harness/src/payload/dogstatsd/common.rs:138]: An accepted edge-case.

File: test/antithesis/harness/src/payload/dogstatsd/common.rs [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+        if i > 0 {
```
[the remark this answers is not in the document]
Comment (blt) [reply test/antithesis/harness/src/payload/dogstatsd/common.rs:139]: An accepted edge-case.

File: test/antithesis/harness/src/bin/parallel_driver_send_dogstatsd.rs [hunk 5]
```diff
@@ -28,81 +25,82 @@ struct Config {
...
-        Cardinality::Medium => "medium",
-        Cardinality::High => "high",
-    };
-    let count: u64 = rng.random_range(50..=2000);
 
-    let socket = connect_with_retry(&config.dogstatsd_socket)?;
+    // Socket unavailable (ADP booting, or a fault). No-op exit, not a failure.
+    let Some(socket) = connect_with_retry(&config.dogstatsd_socket) else {
+        return Ok(());
+    };
```
[the remark this answers is not in the document]
Comment (blt) [reply test/antithesis/harness/src/bin/parallel_driver_send_dogstatsd.rs:45]: This only matters if ADP is kept down for an entire timeline which is not interesting anyway.

**Review by tobz — APPROVED** [review]
